torch/csrc/jit/python/pybind_utils.h

```diff
@@ -496,17 +496,43 @@ inline InferredType tryToInferType(py::handle input) {
   }
 
   // Try container types
-  return tryToInferContainerType(input);
+  return tryToInferContainerType(input, false);
 }
 
-inline InferredType tryToInferContainerType(py::handle input) {
+// This function is similar to tryToInferType, but it only tries to infer
+// primitive types (int, float, bool, complex).
```

Comment:
or container of primitive types, and nesting seems to be allowed?

Reply:
fixed.